### ISSUE TITLE
Softmax GAN Z

### ISSUE THREAD

**Issue by odedbd** [wiseodd/generative-models#17]
Hi,

First I'd like to thank you for this great library of implementation. I find it very helpful in understanding the different variants of GANs.

I think there might be an error in the calculation of the partition function for the Softmax GAN.

[The relevant line](https://github.com/wiseodd/generative-models/blob/master/GAN/softmax_gan/softmax_gan_tensorflow.py#L87) reads:

`Z = tf.reduce_mean(tf.exp(-D_real)) + tf.reduce_mean(tf.exp(-D_fake))`

Where I believe if should have been using `tf.reduce_sum` rather than `tf.reduce_mean`, as the partition function is a sum, not an average. Perhaps this is accounted for in a further part of the code that I am missing.

Thanks,
o

**Comment by wiseodd** [comment]
Hi, thanks for the correction. You're right.
Also, there is a further comment by the author of the paper: https://github.com/wiseodd/generative-models/commit/ea5d40ee613ba0ccc20dae54d0854d25ed7528e8#commitcomment-21891481.

**Comment by odedbd** [comment]
Interesting comment about the stability of the calculation. Please close this issue when you feel it is appropriate. And thanks again for your great library.
